tests/Http/Controllers/ProfileTest.php

```diff
@@ -0,0 +1,434 @@
+        ]);
+
+        $this->actingAs($admin)
+            ->putJson(route('profile.update', $user->id), [
+                'name' => 'Joe Generic',
+                'avatar' => 'https://genericAvatars.com',
+                'about_me' => 'This is a sample bit about me'
+            ])->assertStatus(302);
+
+        $this->assertDatabaseMissing('users', [
+            'id' => $user->id,
+            'name' => 'Joe Generic',
+            'email' => $user->email,
+            'password' => $user->password,
+            'avatar' => 'https://genericAvatars.com',
+            'about_me' => 'This is a sample bit about me',
+            'is_admin' => $user->is_admin
+        ]);
+    }
+
+    /** @test */
+    public function user_can_update_own_profile()
+    {
+        $user = User::factory()->create([
+            'is_admin' => false
+        ]);
+
+        $this->actingAs($user)
+            ->putJson(route('profile.update', $user->id), [
+                'name' => 'Joe Generic',
+                'avatar' => 'https://genericAvatars.com',
+                'about_me' => 'This is a sample bit about me'
+            ])->assertStatus(302);
+
+        $this->assertDatabaseHas('users', [
+            'id' => $user->id,
+            'name' => 'Joe Generic',
+            'email' => $user->email,
+            'password' => $user->password,
+            'avatar' => 'https://genericAvatars.com',
+            'about_me' => 'This is a sample bit about me',
+            'is_admin' => $user->is_admin
+        ]);
+    }
+
+    /** @test */
+    public function user_cannot_update_others_profile()
+    {
+        $user1 = User::factory()->create([
+            'is_admin' => false
+        ]);
+        $user2 = User::factory()->create([
+            'is_admin' => false
+        ]);
+
+        $this->actingAs($user1)
+            ->putJson(route('profile.update', $user2->id), [
+                'name' => 'Joe Generic',
+                'avatar' => 'https://genericAvatars.com',
+                'about_me' => 'This is a sample bit about me'
+            ])->assertStatus(302);
+
+        $this->assertDatabaseMissing('users', [
+            'id' => $user2->id,
+            'name' => 'Joe Generic',
+            'email' => $user2->email,
+            'password' => $user2->password,
+            'avatar' => 'https://genericAvatars.com',
+            'about_me' => 'This is a sample bit about me',
+            'is_admin' => $user2->is_admin
+        ]);
+    }
+
+    /** @test */
+    public function guest_cannot_update_profile()
+    {
+        $user = User::factory()->create([
+            'is_admin' => false
+        ]);
+
+        $this->putJson(route('profile.update', $user->id), [
+                'name' => 'Joe Generic',
+                'avatar' => 'https://genericAvatars.com',
+                'about_me' => 'This is a sample bit about me'
+            ])->assertStatus(401); //unauthorised
+
+        $this->assertDatabaseMissing('users', [
```

Comment:
This kind of works, but when I test update I specify the initial values->then make the request->assert if the database has the same values as before. That way I am certain that the existing record was not deleted/altered.